### DISCUSSION TITLE
Turingpi + Gokrazy

### DISCUSSION THREAD

**Discussion by dev-124** [gokrazy/gokrazy#139] [Ideas]
Hi, 
Interesting project! Really appealing to cut out dependencies like that. 

Thought: Could Gokrazy run on a Turingpi ? 

Wondering if a gokrazy image could be flashed onto the turingpi using the turingpi firmware tools and which parts of Gokrazy and/or Turingpi I need to investigate to determine if this is feasible or will fail because of some cross dependency in these tools ( or perhaps a hardware incompatibility )

Context: after years of using a simple rp2 setup with a few programs. Im looking to level up with a new hw + sw setup for my home with the ability to self host programs directly like this or via docker ( mainly to learn new things) 

Thank you!

**Comment by stapelberg** [comment]
Are you talking about a [Turing Pi V2](https://turingpi.com/turing-pi-v2-is-here/) or the older Turing Pi 1?

In either case, the main obstacle is to figure out how the boot process works.

On the Raspberry Pi, you need to place a few firmware files with the right names on the FAT file system on the first partition of the boot medium (SD card typically), so that the Pi’s bootloader picks them up and starts your kernel.

For other systems, possibly you’d need a special kernel from the vendor’s OS. And, most likely, you’ll need to modify the files on the FAT file system that gokrazy generates to make the Turing Pi bootloader find and understand your config. I can’t easily find on the web how it works, but perhaps you have come across more details on that already?

See https://gokrazy.org/platforms/#community for community-supported gokrazy platforms. There’s also Odroid support that’s done, but not yet documented: https://github.com/gokrazy/gokrazy/issues/101. Maybe either of these are helpful to look things up.

**Reply by dev-124** [reply]
Thank you for laying that out so clearly, there is plenty for me to look into and better understand before moving forward, especially with this multi node + management bus type of setup. Was thinking about the Turing Pi V2 depending on price and availability, but maybe, if it turns out to be too complex right now, I'll probably pick a regular Pi 4.

Can't find v2 docs, in [v1 docs](https://docs.turingpi.com/) there is a mention of flashing the compute modules. So perhaps I would need to [flash those individually](https://docs.turingpi.com/get-started/flashing-compute-modules#4.-boot-the-compute-module). 

Will need to read more, maybe some turing-pi community projects could give a hint
So much to look into :)
